docs/man/man1/bqueues.1
.ds ]W %
.ds ]L
.nh
.TH bqueues 1 "Lava Version 1.0 - Sept 2007"
.br
.SH NAME
\fBbqueues\fR - displays information about queues
.SH SYNOPSIS
.BR
.PP
.PP
\fBbqueues \fR[\fB-w\fR | \fB-l\fR] [\fB-m\fR \fIhost_name\fR\fI \fR|\fI \fR\fB-m\fR\fI host_group\fR\fI | \fR\fB-m all\fR] 
[\fB-u\fR\fB \fR\fIuser_name \fR|\fI \fR\fB-u\fR\fI user_group | \fR\fB-u all\fR] [\fIqueue_name\fR ...] 
.PP
\fBbqueues \fR[\fB-h\fR | \fB-V\fR]
.SH DESCRIPTION
.BR
.PP
.PP
\fB\fRDisplays information about queues.
.PP
By default, returns the following information about all queues: queue 
name, queue priority, queue status, job slot statistics, and job state 
statistics.
.PP
Batch queue names and characteristics are set up by the Lava 
administrator (see lsb.queues(5) and mbatchd(8)).
.SH OPTIONS
.BR
.PP
.TP 
\fB-w\fR 

.IP
Displays queue information in a wide format. Fields are displayed 
without truncation.


.TP 
\fB-l 
\fR
.IP
Displays queue information in a long multi-line format. The -l option 
displays the following additional information: queue description, 
queue characteristics and statistics, scheduling parameters, resource 
limits, scheduling policies, users, hosts, user shares, windows, 
associated commands, and job controls.


.TP 
\fB-m\fR \fIhost_name \fR|\fI \fR\fB-m\fR\fI host_group \fR| \fB-m\fR\fI \fR\fBall\fR 

.IP
Displays the queues that can run jobs on the specified host or host 
group. If the keyword all is specified, displays the queues that can run 
jobs on all hosts . For a list of host groups see bmgroup(1).


.TP 
\fB-u\fR \fIuser_name \fR|\fI \fR\fB-u\fR\fI user_group\fR\fB \fR|\fB -u all\fR 

.IP
Displays the queues that can accept jobs from the specified user or user 
group (For a list of user groups see bugroup(1).) If the keyword `all' 
is specified, displays the queues that can accept jobs from all users.


.TP 
\fIqueue_name \fR... 

.IP
Displays information about the specified queues. 


.TP 
\fB-h\fR 

.IP
Prints command usage to stderr and exits. 


.TP 
\fB-V 
\fR
.IP
Prints Lava release version to stdout and exits. 


.SH OUTPUT
.BR
.PP
.SS Default Output
.BR
.PP
.PP
Displays the following fields: 

.IP
QUEUE_NAME
.BR
.RS
.IP
The name of the queue. Queues are named to correspond to the 
type of jobs usually submitted to them, or to the type of services 
they provide. 


.IP
lost_and_found
.BR
.RS
.IP
If the Lava administrator removes queues from the system, Lava 
creates a queue called lost_and_found and places the jobs 
from the removed queues into the lost_and_found queue. 
Jobs in the lost_and_found queue will not be started unless 
they are switched to other queues (see bswitch).

.RE
.RE
.RE

.IP
PARAMETER/STATISTICS
.BR
.RS

.IP
PRIO
.BR
.RS
.IP
The priority of the queue. If job priority is not configured, 
determines the queue search order at job dispatch, suspension 
and resumption time. Queues with higher priority values are 
searched first for job dispatch and resumption (this is contrary 
to UNIX process priority ordering), and queues with higher 
priority values are searched last for job suspension. 

.RE

.IP
STATUS
.BR
.RS
.IP
The current status of the queue. The possible values are: 


.IP
Open 
.BR
.RS
.IP
The queue is able to accept jobs. 

.RE

.IP
Closed 
.BR
.RS
.IP
The queue is not able to accept jobs. 

.RE

.IP
Active 
.BR
.RS
.IP
Jobs in the queue may be started. 

.RE

.IP
Inactive
.BR
.RS
.IP
Jobs in the queue cannot be started for the time being. 

.RE
.IP
At any moment, each queue is either Open or Closed, and is 
either Active or Inactive. The queue can be opened, closed, 
inactivated and re-activated by the Lava administrator using 
badmin (see badmin(8)). The queue becomes inactive when 
either its dispatch window is closed or its run window is closed 
(see DISPATCH_WINDOWS in the "Output for the -l Option" 
section). In this case, the queue cannot be activated using 
badmin. The queue is re-activated by Lava when one of its 
dispatch windows and one of its run windows are open again. 
The initial state of a queue at Lava boot time is set to open, and 
either active or inactive depending on its windows. 

.RE
.RE

.IP
MAX
.BR
.RS
.IP
The maximum number of job slots that can be used by the jobs 
from the queue. These job slots are used by dispatched jobs 
which have not yet finished, and by pending jobs which have 
slots reserved for them. A sequential job will use one job slot 
when it is dispatched to a host, while a parallel job will use as 
many job slots as is required by bsub -n when it is dispatched. 
See bsub(1) for details. If `-' is displayed, there is no limit. 

.RE

.IP
JL/U
.BR
.RS
.IP
The maximum number of job slots you can use for your jobs in 
the queue. These job slots are used by your dispatched jobs 
which have not yet finished, and by pending jobs which have 
slots reserved for them. If `-' is displayed, there is no limit. 

.RE

.IP
JL/P 
.BR
.RS
.IP
The maximum number of job slots a processor can process 
from the queue. This includes job slots of dispatched jobs that 
have not yet finished, and job slots reserved for some pending 
jobs. The job slot limit per processor (JL/P) controls the number 
of jobs sent to each host. This limit is configured per processor 
so that multiprocessor hosts are automatically allowed to run 
more jobs. If `-' is displayed, there is no limit. 

.RE

.IP
JL/H
.BR
.RS
.IP
The maximum number of job slots a host can process from the 
queue. This includes the job slots of dispatched jobs that have 
not yet finished, and those reserved for some pending jobs. The 
job slot limit per host (JL/H) controls the number of jobs sent 
to each host, regardless of whether a host is a uniprocessor host 
or a multiprocessor host. If `-' is displayed, there is no limit. 

.RE

.IP
NJOBS 
.BR
.RS
.IP
The total number of job slots held currently by jobs in the 
queue. This includes pending, running, suspended and 
reserved job slots. A parallel job that is running on \fIn\fR processors 
is counted as \fIn\fR job slots, since it takes \fIn\fR job slots in the queue. 
See bjobs(1) for an explanation of batch job states. 

.RE

.IP
PEND 
.BR
.RS
.IP
The number of pending job slots in the queue. 

.RE

.IP
RUN
.BR
.RS
.IP
The number of running job slots in the queue. 

.RE

.IP
SUSP
.BR
.RS
.IP
The number of suspended job slots in the queue.

.RE
.SS Output for \fB-\fRl Option
.BR
.PP
.PP
In addition to the above fields, the \fB-\fRl option displays the following: 

.IP
Description 
.BR
.RS
.IP
A description of the typical use of the queue. 

.RE

.IP
PARAMETERS/STATISTICS
.BR
.RS

.IP
NICE 
.BR
.RS
.IP
The nice value at which jobs in the queue will be run. This is 
the UNIX nice value for reducing the process priority (see 
nice(1)). 

.RE

.IP
STATUS
.BR
.RS

.IP
Inactive
.BR
.RS
.IP
The long format for the \fB-\fRl option gives the possible reasons 
for a queue to be inactive:

.RE

.IP
Inact_Win
.BR
.RS
.IP
The queue is out of its dispatch window or its run window.

.RE

.IP
Inact_Adm
.BR
.RS
.IP
The queue has been inactivated by the Lava administrator.

.RE

.IP
SSUSP
.BR
.RS
.IP
The number of job slots in the queue allocated to jobs that are 
suspended by Lava.

.RE

.IP
USUSP 
.BR
.RS
.IP
The number of job slots in the queue allocated to jobs that are 
suspended by the job submitter or by the Lava administrator.

.RE

.IP
RSV 
.BR
.RS
.IP
The numbers of job slots in the queue that are reserved by Lava 
for pending jobs.

.RE

.IP
Migration threshold
.BR
.RS
.IP
\fB\fRThe length of time in seconds that a job dispatched from the queue 
will remain suspended by the system before Lava attempts to 
migrate the job to another host. See the MIG parameter in 
lsb.queues and lsb.hosts.

.RE

.IP
Schedule delay for a new job
.BR
.RS
.IP
The delay time in seconds for scheduling a session after a new job 
is submitted. If the schedule delay time is zero, a new scheduling 
session is started as soon as the job is submitted to the queue. See 
the NEW_JOB_SCHEDULE_DELAY parameter in lsb.queues.

.RE

.IP
Interval for a host to accept two jobs
.BR
.RS
.IP
The length of time in seconds to wait after dispatching a job to a 
host before dispatching a second job to the same host. If the job 
accept interval is zero, a host may accept more than one job in each 
dispatching interval. See the JOB_ACCEPT_INTERVAL parameter in 
lsb.queues and lsb.params.

.RE

.IP
RESOURCE LIMITS 
.BR
.RS
.IP
The hard resource limits that are imposed on the jobs in the queue 
(see getrlimit(2) and lsb.queues(5)). These limits are imposed 
on a per-job and a per-process basis. 

.IP
The possible per-job limits are: 

.IP
CPULIMIT

.IP
PROCLIMIT

.IP
MEMLIMIT

.IP
SWAPLIMIT

.IP
The possible UNIX per-process resource limits are:

.IP
RUNLIMIT

.IP
FILELIMIT

.IP
DATALIMIT

.IP
STACKLIMIT

.IP
CORELIMIT

.IP
If a job submitted to the queue has any of these limits specified (see 
bsub(1)), then the lower of the corresponding job limits and 
queue limits are used for the job. 

.IP
If no resource limit is specified, the resource is assumed to be 
unlimited. 

.RE

.IP
SCHEDULING PARAMETERS 
.BR
.RS
.IP
The scheduling and suspending thresholds for the queue. 

.IP
The scheduling threshold loadSched and the suspending 
threshold loadStop are used to control batch job dispatch, 
suspension, and resumption. The queue thresholds are used in 
combination with the thresholds defined for hosts (see bhosts(1) 
and lsb.hosts(5)). If both queue level and host level thresholds 
are configured, the most restrictive thresholds are applied.

.IP
The loadSched and loadStop thresholds have the following 
fields: 


.IP
r15s 
.BR
.RS
.IP
The 15-second exponentially averaged effective CPU run queue 
length.

.RE

.IP
r1m
.BR
.RS
.IP
The 1-minute exponentially averaged effective CPU run queue 
length. 

.RE

.IP
r15m
.BR
.RS
.IP
The 15-minute exponentially averaged effective CPU run queue 
length. 

.RE

.IP
ut
.BR
.RS
.IP
The CPU utilization exponentially averaged over the last 
minute, expressed as a percentage between 0 and 1. 

.RE

.IP
pg
.BR
.RS
.IP
The memory paging rate exponentially averaged over the last 
minute, in pages per second. 

.RE

.IP
io
.BR
.RS
.IP
The disk I/O rate exponentially averaged over the last minute, 
in kilobytes per second. 

.RE

.IP
ls 
.BR
.RS
.IP
The number of current login users. 

.RE

.IP
it
.BR
.RS
.IP
On UNIX, the idle time of the host (keyboard not touched on 
all logged in sessions), in minutes.
.RE

.IP
tmp 
.BR
.RS
.IP
The amount of free space in /tmp, in megabytes. 

.RE

.IP
swp 
.BR
.RS
.IP
The amount of currently available swap space, in megabytes. 

.RE

.IP
mem 
.BR
.RS
.IP
The amount of currently available memory, in megabytes. 

.RE
.IP
In addition to these internal indices, external indices are also 
displayed if they are defined in lsb.queues (see lsb.queues(5)). 

.IP
The loadSched threshold values specify the job dispatching 
thresholds for the corresponding load indices. If `-' is displayed as 
the value, it means the threshold is not applicable. Jobs in the 
queue may be dispatched to a host if the values of all the load 
indices of the host are within (below or above, depending on the 
meaning of the load index) the corresponding thresholds of the 
queue and the host. The same conditions are used to resume jobs 
dispatched from the queue that have been suspended on this host. 

.IP
Similarly, the loadStop threshold values specify the thresholds for 
job suspension. If any of the load index values on a host go beyond 
the corresponding threshold of the queue, jobs in the queue will 
be suspended. 

.RE
.RE

.IP
SCHEDULING POLICIES
.BR
.RS
.IP
Scheduling policies of the queue. Optionally, one or more of the 
following policies may be configured: 


.IP
IGNORE_DEADLINE
.BR
.RS
.IP

src/lsbatch/cmd/bqueues.c
/* $Id: bqueues.c,v 1.8 2007/08/15 22:18:43 tmizan Exp $
 * Copyright (C) 2007 Platform Computing Inc
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of version 2 of the GNU General Public License as
 * published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.

 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
 *
 */

#include "lsbatch/cmd/cmd.h"

void load2Str();
static void prtQueuesLong (int, struct queueInfoEnt *);
static void prtQueuesShort (int, struct queueInfoEnt *);
static char wflag = FALSE;
extern int terminateWhen_(int *, char *);

#define QUEUE_NAME_LENGTH    15
#define QUEUE_PRIO_LENGTH    4
#define QUEUE_STATUS_LENGTH  14
#define QUEUE_JL_U_LENGTH    4
#define QUEUE_JL_P_LENGTH    4
#define QUEUE_JL_H_LENGTH    4
#define QUEUE_MAX_LENGTH     4
#define QUEUE_NJOBS_LENGTH   5
#define QUEUE_PEND_LENGTH    5
#define QUEUE_RUN_LENGTH     5
#define QUEUE_SUSP_LENGTH    5
#define QUEUE_SSUSP_LENGTH   5
#define QUEUE_USUSP_LENGTH   5
#define QUEUE_NICE_LENGTH    4
#define QUEUE_RSV_LENGTH     4

static char fomt[200];

void
usage (char *cmd)
{
    fprintf(stderr, "Usage");
    fprintf(stderr, ": %s [-h] [-V] [-w | -l] [-m host_name | -m cluster_name]\n", cmd);

    if (lsbMode_ & LSB_MODE_BATCH)
	fprintf(stderr, " [-u user_name]");
    fprintf(stderr, " [queue_name ...]\n");
    exit(-1);
}

int
main (int argc, char **argv)
{
    extern int optind;
    extern char *optarg;
    int numQueues;
    char **queueNames=NULL, **queues = NULL;
    struct queueInfoEnt *queueInfo;
    char lflag = FALSE;
    int cc, defaultQ = FALSE;
    char *host = NULL, *user = NULL;


    numQueues = 0;

    if (lsb_init(argv[0]) < 0) {
	lsb_perror("lsb_init");
	exit(-1);
    }

    while ((cc = getopt(argc, argv, "Vhlwm:u:")) != EOF) {
        switch (cc) {
        case 'l':
            lflag = TRUE;
            if (wflag)
                usage(argv[0]);
            break;
        case 'w':
            wflag = TRUE;
            if (lflag)
                usage(argv[0]);
            break;
        case 'm':
            if (host != NULL || *optarg == '\0')
                usage(argv[0]);
            host = optarg;
            break;
        case 'u':
            if (user != NULL || *optarg == '\0')
                usage(argv[0]);
            user = optarg;
            break;
	case 'V':
	    fprintf(stderr, "%s\n", LAVALITE_VERSION_STR);
	    exit(0);
        case 'h':
        default:
            usage(argv[0]);
        }
    }

    numQueues = getNames (argc, argv, optind, &queueNames,  &defaultQ, "queue");
    if (!defaultQ && numQueues != 0)
        queues = queueNames;
    else
        queues = NULL;

    TIMEIT(0, (queueInfo = lsb_queueinfo(queues,
					 &numQueues,
					 host,
					 user,
					 0)), "lsb_queueinfo");

    if (!queueInfo) {
        if (lsberrno == LSBE_BAD_QUEUE && queues)
	    lsb_perror(queues[numQueues]);
        else {
            switch (lsberrno) {
            case LSBE_BAD_HOST   :
            case LSBE_QUEUE_HOST :
                lsb_perror (host);
                break;
            case LSBE_BAD_USER   :
            case LSBE_QUEUE_USE  :
                lsb_perror (user);
                break;
            default :
                lsb_perror (NULL);
            }
         }
         exit(-1);
    }
    if (lflag )
        prtQueuesLong(numQueues, queueInfo);
    else
        prtQueuesShort(numQueues, queueInfo);
    exit(0);

}

static void
prtQueuesLong(int numQueues, struct queueInfoEnt *queueInfo)
{
    struct queueInfoEnt*  qp;
    char                  statusStr[64];
    char                  userJobLimit[MAX_CHARLEN];
    char                  procJobLimit[MAX_CHARLEN];
    char                  hostJobLimit[MAX_CHARLEN];
    char                  maxJobs[MAX_CHARLEN];
    int                   i;
    int                   numDefaults = 0;
    struct lsInfo*        lsInfo;
    int                   printFlag = 0;
    int                   printFlag1 = 0;
    int                   printFlag2 = 0;
    int			  procLimits[3];

    if ((lsInfo = ls_info()) == NULL) {
	ls_perror("ls_info");
	exit(-1);
    }

    printf("\n");

    for ( i=0; i<numQueues; i++ )  {
	 qp = &(queueInfo[i]);
	 if (qp->qAttrib & Q_ATTRIB_DEFAULT)
	     numDefaults++;
    }

    for ( i=0; i<numQueues; i++ ) {

        qp = &(queueInfo[i]);

        if (qp->qStatus & QUEUE_STAT_OPEN) {
            sprintf(statusStr, "%s:", "Open");
	}
	else {
            sprintf(statusStr, "%s:", "Closed");
	}
        if (qp->qStatus & QUEUE_STAT_ACTIVE) {
            if (qp->qStatus & QUEUE_STAT_RUN) {
	        strcat(statusStr, "Active");
	    } else {
	        strcat(statusStr, I18N_Inact__Win);
	    }
	} else {
	    strcat(statusStr, I18N_Inact__Adm);
	}

	if (qp->maxJobs < INFINIT_INT)
            strcpy(maxJobs, prtValue(QUEUE_MAX_LENGTH, qp->maxJobs) );
        else
            strcpy(maxJobs, prtDash(QUEUE_MAX_LENGTH) );

	if (qp->userJobLimit < INFINIT_INT)
            strcpy(userJobLimit,
                   prtValue(QUEUE_JL_U_LENGTH, qp->userJobLimit) );
        else
            strcpy(userJobLimit, prtDash(QUEUE_JL_U_LENGTH) );

        if (qp->procJobLimit < INFINIT_FLOAT) {
            sprintf(fomt, "%%%d.1f ", QUEUE_JL_P_LENGTH);
            sprintf (procJobLimit, fomt, qp->procJobLimit);
	}
        else
            strcpy(procJobLimit, prtDash(QUEUE_JL_P_LENGTH) );

        if (qp->hostJobLimit < INFINIT_INT)
            strcpy(hostJobLimit,
                   prtValue(QUEUE_JL_H_LENGTH, qp->hostJobLimit) );
        else
            strcpy(hostJobLimit, prtDash(QUEUE_JL_H_LENGTH) );

        if (i > 0)
            printf("-------------------------------------------------------------------------------\n\n");
        printf("%s: %s\n", "QUEUE", qp->queue);

        printf("  -- %s", qp->description);
	if (qp->qAttrib & Q_ATTRIB_DEFAULT) {
            if (numDefaults == 1)
	       printf("  %s.\n\n", "This is the default queue");
            else
	       printf("  %s.\n\n", "This is one of the default queues");
        } else
	    printf("\n\n");

        printf(("PARAMETERS/STATISTICS\n"));

        prtWord(QUEUE_PRIO_LENGTH, "PRIO", 0);

	if ( lsbMode_ & LSB_MODE_BATCH )
            prtWord(QUEUE_NICE_LENGTH, "NICE", 1);

        prtWord(QUEUE_STATUS_LENGTH, "STATUS", 0);

	if ( lsbMode_ & LSB_MODE_BATCH ) {
            prtWord(QUEUE_MAX_LENGTH,  "MAX", -1);
            prtWord(QUEUE_JL_U_LENGTH, I18N_JL_U, -1);
            prtWord(QUEUE_JL_P_LENGTH, I18N_JL_P, -1);
            prtWord(QUEUE_JL_H_LENGTH, I18N_JL_H, -1);
	};

        prtWord(QUEUE_NJOBS_LENGTH, "NJOBS", -1);
        prtWord(QUEUE_PEND_LENGTH,  "PEND",  -1);
        prtWord(QUEUE_RUN_LENGTH,   "RUN",   -1);
        prtWord(QUEUE_SSUSP_LENGTH, "SSUSP", -1);
        prtWord(QUEUE_USUSP_LENGTH, "USUSP", -1);
        prtWord(QUEUE_RSV_LENGTH,   "RSV",   -1);
        printf("\n");

        prtWordL(QUEUE_PRIO_LENGTH,
                 prtValue(QUEUE_PRIO_LENGTH-1, qp->priority));

	if ( lsbMode_ & LSB_MODE_BATCH )
            prtWordL(QUEUE_NICE_LENGTH,
                     prtValue(QUEUE_NICE_LENGTH-1, qp->nice));

        prtWord(QUEUE_STATUS_LENGTH, statusStr, 0);

	if ( lsbMode_ & LSB_MODE_BATCH ) {
            sprintf(fomt, "%%%ds%%%ds%%%ds%%%ds", QUEUE_MAX_LENGTH,
                                                  QUEUE_JL_U_LENGTH,
                                                  QUEUE_JL_P_LENGTH,
                                                  QUEUE_JL_H_LENGTH );
	    printf(fomt,
                   maxJobs, userJobLimit, procJobLimit, hostJobLimit);
	};

        sprintf(fomt, "%%%dd %%%dd %%%dd %%%dd %%%dd %%%dd\n",
                                                  QUEUE_NJOBS_LENGTH,
                                                  QUEUE_PEND_LENGTH,
                                                  QUEUE_RUN_LENGTH,
                                                  QUEUE_SSUSP_LENGTH,
                                                  QUEUE_USUSP_LENGTH,
                                                  QUEUE_RSV_LENGTH );
	printf(fomt,
                qp->numJobs, qp->numPEND, qp->numRUN,
                qp->numSSUSP, qp->numUSUSP, qp->numRESERVE);

	if ( qp->mig < INFINIT_INT )
       	    printf(("Migration threshold is %d minutes\n"), qp->mig);

        if ( qp->schedDelay < INFINIT_INT )
	    printf(("Schedule delay for a new job is %d seconds\n"),
		    qp->schedDelay);

        if ( qp->acceptIntvl < INFINIT_INT )
	    printf(("Interval for a host to accept two jobs is %d seconds\n"),
		    qp->acceptIntvl);

        if (((qp->defLimits[LSF_RLIMIT_CPU] != INFINIT_INT) &&
             (qp->defLimits[LSF_RLIMIT_CPU] > 0 )) ||
            ((qp->defLimits[LSF_RLIMIT_RUN] != INFINIT_INT) &&
             (qp->defLimits[LSF_RLIMIT_RUN] > 0 )) ||
            ((qp->defLimits[LSF_RLIMIT_DATA] != INFINIT_INT) &&
             (qp->defLimits[LSF_RLIMIT_DATA] > 0 )) ||
            ((qp->defLimits[LSF_RLIMIT_RSS] != INFINIT_INT) &&
             (qp->defLimits[LSF_RLIMIT_RSS] > 0 )) ||
            ((qp->defLimits[LSF_RLIMIT_PROCESS] != INFINIT_INT) &&
             (qp->defLimits[LSF_RLIMIT_PROCESS] > 0 ))) {

	     printf("\n");
	     printf("DEFAULT LIMITS:"  );
	     prtResourceLimit (qp->defLimits, qp->hostSpec, 1.0, 0);
	     printf("\n");
	     printf("MAXIMUM LIMITS:"  );
	}

	procLimits[0] = qp->minProcLimit;
	procLimits[1] = qp->defProcLimit;
	procLimits[2] = qp->procLimit;
        prtResourceLimit (qp->rLimits, qp->hostSpec, 1.0, procLimits);

	if ( lsbMode_ & LSB_MODE_BATCH ) {
	    printf(("\nSCHEDULING PARAMETERS\n"));

 	    if (printThresholds (qp->loadSched,  qp->loadStop, NULL, NULL,
				 MIN(lsInfo->numIndx, qp->nIdx), lsInfo) < 0)
		exit (-1);
	}

        if ((qp->qAttrib & Q_ATTRIB_EXCLUSIVE)
            || (qp->qAttrib & Q_ATTRIB_BACKFILL)
            || (qp->qAttrib & Q_ATTRIB_IGNORE_DEADLINE)
            || (qp->qAttrib & Q_ATTRIB_ONLY_INTERACTIVE)
	    || (qp->qAttrib & Q_ATTRIB_NO_INTERACTIVE)) {
            printf("\n%s:", "SCHEDULING POLICIES");
            if (qp->qAttrib & Q_ATTRIB_BACKFILL)
                printf("  %s", ("BACKFILL"));
            if (qp->qAttrib & Q_ATTRIB_IGNORE_DEADLINE)
                printf("  %s", ("IGNORE_DEADLINE"));
            if (qp->qAttrib & Q_ATTRIB_EXCLUSIVE)
                printf("  %s", ("EXCLUSIVE"));
            if (qp->qAttrib & Q_ATTRIB_NO_INTERACTIVE)
                printf("  %s", ("NO_INTERACTIVE"));
            if (qp->qAttrib & Q_ATTRIB_ONLY_INTERACTIVE)
                printf("  %s", ("ONLY_INTERACTIVE"));
	    printf("\n");
        }

        if (strcmp (qp->defaultHostSpec, " ") !=  0)
            printf("\n%s: %s\n",
		("DEFAULT HOST SPECIFICATION"), qp->defaultHostSpec);

        if (qp->windows && strcmp (qp->windows, " " ) !=0)
	    printf("\n%s: %s\n",
		("RUN_WINDOW"), qp->windows);
        if (strcmp (qp->windowsD, " ")  !=  0)
	    printf("\n%s: %s\n",
		("DISPATCH_WINDOW"), qp->windowsD);

	if (lsbMode_ & LSB_MODE_BATCH) {
	    if ( strcmp(qp->userList, " ") == 0) {
		printf("\n%s:  %s\n", "USERS",
		       "all users");
	    } else {
		if (strcmp(qp->userList, " ") != 0 && qp->userList[0] != 0)
		    printf("\n%s: %s\n", "USERS", qp->userList);
	    }
	}

	if (strcmp(qp->hostList, " ") == 0) {
	    if (lsbMode_ & LSB_MODE_BATCH)
		printf("%s\n",
			("HOSTS:  all hosts used by the LSF Batch system"));
	    else
		printf("%s\n",
			("HOSTS: all hosts used by the LSF JobScheduler system"));
        } else {
	    if (strcmp(qp->hostList, " ") != 0 && qp->hostList[0])
	        printf("%s:  %s\n", "HOSTS", qp->hostList);
        }
        if (strcmp (qp->admins, " ") != 0)
            printf("%s:  %s\n",
		("ADMINISTRATORS"), qp->admins);
        if (strcmp (qp->preCmd, " ") != 0)
            printf("%s:  %s\n",
		("PRE_EXEC"), qp->preCmd);
        if (strcmp (qp->postCmd, " ") != 0)
            printf("%s:  %s\n",
		("POST_EXEC"), qp->postCmd);
        if (strcmp (qp->requeueEValues, " ") != 0)
            printf("%s:  %s\n",
		("REQUEUE_EXIT_VALUES"), qp->requeueEValues);
        if (strcmp (qp->resReq, " ") != 0)
            printf("%s:  %s\n",
		("RES_REQ"), qp->resReq);
        if (qp->slotHoldTime > 0)
	    printf(("Maximum slot reservation time: %d seconds\n"), qp->slotHoldTime);
        if (strcmp (qp->resumeCond, " ") != 0)
	    printf("%s:  %s\n",
		("RESUME_COND"), qp->resumeCond);
        if (strcmp (qp->stopCond, " ") != 0)
	    printf("%s:  %s\n",
		("STOP_COND"), qp->stopCond);
        if (strcmp (qp->jobStarter, " ") != 0)
	    printf("%s:  %s\n",
		("JOB_STARTER"), qp->jobStarter);
	if ( qp->qAttrib & Q_ATTRIB_RERUNNABLE )
	   printf(("RERUNNABLE :  yes\n"));

	if ( qp->qAttrib & Q_ATTRIB_CHKPNT ) {
	   printf(("CHKPNTDIR : %s\n"), qp->chkpntDir);
	   printf(("CHKPNTPERIOD : %d\n"), qp->chkpntPeriod);
	}

        printf("\n");
        printFlag = 0;
        if  ((qp->suspendActCmd != NULL)
            && (qp->suspendActCmd[0] != ' '))
                printFlag = 1;

        printFlag1 = 0;
        if  ((qp->resumeActCmd != NULL)
            && (qp->resumeActCmd[0] != ' '))
            printFlag1 = 1;

        printFlag2 = 0;
        if  ((qp->terminateActCmd != NULL)
            && (qp->terminateActCmd[0] != ' '))
            printFlag2 = 1;

        if (printFlag || printFlag1 || printFlag2)
            printf("%s:\n",
		("JOB_CONTROLS"));

        if (printFlag) {
            printf("    %-9.9s", ("SUSPEND:"));
            if (strcmp (qp->suspendActCmd, " ") != 0)
                printf("    [%s]\n", qp->suspendActCmd);
        }

        if (printFlag1) {
            printf("    %-9.9s", ("RESUME:"));
            if (strcmp (qp->resumeActCmd, " ") != 0)
                printf("    [%s]\n", qp->resumeActCmd);
        }

        if (printFlag2) {
            printf("    %-9.9s", ("TERMINATE:"));
            if (strcmp (qp->terminateActCmd, " ") != 0)
                printf("    [%s]\n", qp->terminateActCmd);
        }

        if (printFlag || printFlag1 || printFlag2)
            printf("\n");

        printFlag = terminateWhen_(qp->sigMap, "USER");
        printFlag1 = terminateWhen_(qp->sigMap, "WINDOW");
        printFlag2 = terminateWhen_(qp->sigMap, "LOAD");

        if (printFlag | printFlag1 | printFlag2) {
            printf(("TERMINATE_WHEN = "));
            if (printFlag) printf(("USER "));
            if (printFlag1) printf(("WINDOW "));
            if (printFlag2) printf(("LOAD"));
	    printf("\n");
        };

    };

    printf("\n");

}
static void
prtQueuesShort(int numQueues, struct queueInfoEnt *queueInfo)
{
    struct queueInfoEnt *qp;
    char statusStr[64];
    char first = FALSE;
    int i;
    char userJobLimit[MAX_CHARLEN],
         procJobLimit[MAX_CHARLEN],
         hostJobLimit[MAX_CHARLEN];
    char maxJobs[MAX_CHARLEN];

    if( !first ) {
            prtWord(QUEUE_NAME_LENGTH, I18N_QUEUE__NAME, 0);
            prtWord(QUEUE_PRIO_LENGTH, "PRIO", 1);
            prtWord(QUEUE_STATUS_LENGTH, "STATUS", 0);

	    if ( lsbMode_ & LSB_MODE_BATCH ) {
                prtWord(QUEUE_MAX_LENGTH,  "MAX",  -1);
                prtWord(QUEUE_JL_U_LENGTH, I18N_JL_U, -1);
                prtWord(QUEUE_JL_P_LENGTH, I18N_JL_P, -1);
                prtWord(QUEUE_JL_H_LENGTH, I18N_JL_H, -1);
	    };

        prtWord(QUEUE_NJOBS_LENGTH, "NJOBS", -1);
        prtWord(QUEUE_PEND_LENGTH,  "PEND",  -1);
        prtWord(QUEUE_RUN_LENGTH,   "RUN",   -1);
        prtWord(QUEUE_SUSP_LENGTH,  "SUSP",  -1);
        printf("\n");
        first = TRUE;
    }

    for ( i=0; i<numQueues; i++) {

        qp = &(queueInfo[i]);

        if (qp->qStatus & QUEUE_STAT_OPEN) {
            sprintf(statusStr, "%s:", "Open");
	}
	else {
            sprintf(statusStr, "%s:", "Closed");
	}
        if (qp->qStatus & QUEUE_STAT_ACTIVE) {
            if (qp->qStatus & QUEUE_STAT_RUN) {
	        strcat(statusStr, "Active");
	    } else {
	        strcat(statusStr, "Inact");
	    }
	} else {
	    strcat(statusStr, "Inact");
	}

	if (qp->maxJobs < INFINIT_INT)
            strcpy(maxJobs, prtValue(QUEUE_MAX_LENGTH, qp->maxJobs) );
        else
            strcpy(maxJobs, prtDash(QUEUE_MAX_LENGTH) );

	if (qp->userJobLimit < INFINIT_INT)
            strcpy(userJobLimit,
                   prtValue(QUEUE_JL_U_LENGTH, qp->userJobLimit) );
        else
            strcpy(userJobLimit, prtDash(QUEUE_JL_U_LENGTH) );

        if (qp->procJobLimit < INFINIT_FLOAT) {
            sprintf(fomt, "%%%d.0f ", QUEUE_JL_P_LENGTH);
            sprintf (procJobLimit, fomt, qp->procJobLimit);
	}
        else
            strcpy(procJobLimit, prtDash(QUEUE_JL_P_LENGTH) );

        if (qp->hostJobLimit < INFINIT_INT)
            strcpy(hostJobLimit,
                   prtValue(QUEUE_JL_H_LENGTH, qp->hostJobLimit) );
        else
            strcpy(hostJobLimit, prtDash(QUEUE_JL_H_LENGTH) );

        if ( wflag ) {
            prtWordL(QUEUE_NAME_LENGTH, qp->queue);
            prtWordL(QUEUE_PRIO_LENGTH,
                   prtValue(3, qp->priority));
            prtWordL(QUEUE_STATUS_LENGTH, statusStr);
	} else {
            prtWord(QUEUE_NAME_LENGTH, qp->queue, 0);
            prtWord(QUEUE_PRIO_LENGTH,
                   prtValue(3, qp->priority), 1);
            prtWord(QUEUE_STATUS_LENGTH, statusStr, 1);
	};

	if ( lsbMode_ & LSB_MODE_BATCH ) {
            sprintf(fomt, "%%%ds%%%ds%%%ds%%%ds", QUEUE_MAX_LENGTH,
                                                  QUEUE_JL_U_LENGTH,
                                                  QUEUE_JL_P_LENGTH,
                                                  QUEUE_JL_H_LENGTH );
	    printf(fomt,
                   maxJobs, userJobLimit, procJobLimit, hostJobLimit);
	};

        sprintf(fomt, "%%%dd %%%dd %%%dd %%%dd\n", QUEUE_NJOBS_LENGTH,
                                                   QUEUE_PEND_LENGTH,
                                                   QUEUE_RUN_LENGTH,
                                                   QUEUE_SUSP_LENGTH );
	printf(fomt,
                qp->numJobs, qp->numPEND, qp->numRUN,
                (qp->numSSUSP + qp->numUSUSP) );

    };

}
